### DISCUSSION TITLE
Column Type Change - Data Disappear

### DISCUSSION THREAD

**Discussion by yurijserrano** [RafaelGB/obsidian-db-folder#266] [General]
**Describe the bug**
When I change the column type the data disappears

**To Reproduce**
Just change the type of the column and the data will disappear

**Expected behavior**
The data not disappear.

**Video**

https://user-images.githubusercontent.com/14169134/185154493-46798e51-ba4e-4832-bf2a-18194c6654cf.mp4

**Desktop (please complete the following information):**
 - OS: Pop OS
 - Version: 22.04

**Comment by RafaelGB** [comment]
when parsing a text type column to date, since they are not compatible with each other, a date is not shown, but the data remains intact in the files until it is modified

**Comment by yurijserrano** [comment]
Hi Rafael

Correct I could verify that

However I saw that the data is populated correctly when I exit the file and come back again, so I thought that this update can be done when the column type conversion is done.

This way I think the workflow would be better.

An example of this can be seen in the video below:


https://user-images.githubusercontent.com/14169134/185158631-9535c895-2cc2-4b60-886d-f167d3269a5b.mp4

**Reply by RafaelGB** [reply]
hmm during the change it would be to see if, in case the text can be converted into a date, generate that date

**Reply by RafaelGB** [reply]
I will check if this function is working correctly
```ts
private parseToCalendar(wrapped: WrappedLiteral): DateTime {
        if (DateTime.isDateTime(wrapped.value)) {
            if (wrapped.type === 'string') {

                return DateTime.fromISO(wrapped.value);
            } else {
                return wrapped.value;
            }
        } else {
            return null;
        }
    }
```

**Reply by RafaelGB** [reply]
An improved parser:
```ts
private parseToCalendar(wrapped: WrappedLiteral): DateTime {
        if (wrapped.type === 'string') {
            const calendarCandidate = DateTime.fromISO(wrapped.value);
            if (calendarCandidate.isValid) {
                return calendarCandidate;
            }
            return null;
        }

        if (DateTime.isDateTime(wrapped.value)) {
            return wrapped.value;
        } else {
            return null;
        }
    }
```

https://user-images.githubusercontent.com/11924043/185171610-9d403911-3c6b-4f4c-8846-d87268df3de3.mov

**Reply by yurijserrano** [reply]
@RafaelGB Perfect this ends up fixing the problem of the data disappearing.

Do you intend to include this modification in the next version of the DB Folder?

**Reply by RafaelGB** [reply]
yeah Its already included on the main branch. will be added this week with a minor version

**Reply by yurijserrano** [reply]
Thank you so much @RafaelGB  for your attention and fix to this matter

I wish you a great day

**Comment by RafaelGB** [comment]
included with 2.2.1

**Reply by yurijserrano** [reply]
Thank you so much @RafaelGB 

I will be checking this new update later.


I wish you a great day.
